@types/APIClients.ts

```diff
@@ -1,94 +1,126 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-interface ProgressPromise<T> extends Promise<T> {
-    progress?: (
-        callback: (fractionCompleted: number) => void
-    ) => ProgressPromise<T>;
-    onProgress?: (fractionCompleted: number) => void;
-    cancel?: () => void;
-}
+import type { EmitterSubscription } from "react-native";
 
-type ClientHeaders = Record<string, string>;
-
-type RequestOptions = {
-    headers?: ClientHeaders;
-    body?: Record<string, unknown> | string;
-    timeoutInterval?: number;
-    retryPolicyConfiguration?: RetryPolicyConfiguration;
-};
-
-type UploadRequestOptions = RequestOptions & {
-    skipBytes?: number;
-};
-
-type ClientResponse = {
-    headers?: ClientHeaders;
-    data?: Record<string, unknown>;
-    code: number;
-    lastRequestedUrl: string;
-    ok: boolean;
-};
-
-interface GenericClientInterface {
-    get(url: string, options?: RequestOptions): Promise<ClientResponse>;
-    put(url: string, options?: RequestOptions): Promise<ClientResponse>;
-    post(url: string, options?: RequestOptions): Promise<ClientResponse>;
-    patch(url: string, options?: RequestOptions): Promise<ClientResponse>;
-    delete(url: string, options?: RequestOptions): Promise<ClientResponse>;
-}
+declare global {
```

Comment:
I actually just had a thought around this - how about we declare `EmitterSubscription` and `NativeModule` as a global type in a `@types/Global.ts`, and that way we can use both of these without needing to import into our actual declarations.

Something like:

```
// @types/Global.ts

import type { NativeModule as NM, EventEmitter as EE } from "react-native";

declare global {
    interface NativeModule extends NM {}
    interface EventEmitter extends EE {}
}
```

This way we can use EventEmitter and NativeModule globally without needing to import the react types, thus not needing to do `declare global { ....` on our types.